packages/frontend-2/lib/viewer/composables/activity.ts

```diff
@@ -78,6 +88,57 @@ graphql(`
   }
 `)
 
+const useViewerRealtimeActivityState = () =>
+  useState('viewer_realtime_activity_state', () => ({
+    activity: undefined as Optional<ViewerActivityMetadata>,
+    status: ViewerUserActivityStatus.Viewing as ViewerUserActivityStatus
+  }))
+
+export const useViewerRealtimeActivityTracker = () => {
```

Comment:
serializing the viewer state and comparing if its the same as a previous one are relatively expensive operations, ones we already try to do on every change to the viewer state for activity tracking.

so i made sure that collected activity is stored in a global singleton state, one accessible through this new composable anywhere in the viewer, so that you can easily get the state and its unique ID (changes w/ every change to the state) anywhere without having to re-serialize

this will be useful whenever you want to track if there's a change from some previously established state - e.g. you want to see if the user "exited" the saved view by making any changes to the viewport, and you can do it by comparing the state ids